Assets/ModScripts/Submodules/AVInput.cs

```diff
@@ -0,0 +1,239 @@
+    public AVInput(CruelModkitScript Module, int ModuleID, ComponentInfo Info, byte Components) : base(Module, ModuleID, Info, Components)
+    {
+        Debug.LogFormat("[The Cruel Modkit #{0}] Solving AV Input.", ModuleID);
+
+        bulbStates = new bool[2] { Info.BulbInfo[2], Info.BulbInfo[3] };
+        for (int i = 0; i < 5; i++)
+        {
+            int note = Random.Range(0, 12);
+            while (bulb1Notes.Contains(note))
+                note = Random.Range(0, 12);
+
+            bulb1Notes.Add(note);
+            bulb1Actions[note] = 2;
+
+            note = Random.Range(0, 12);
+            while (bulb2Notes.Contains(note))
+                note = Random.Range(0, 12);
+
+            bulb2Notes.Add(note);
+            bulb2Actions[note] = 2;
+        }
+        bulb1Notes.Sort();
+        bulb2Notes.Sort();
+        Debug.LogFormat("[The Cruel Modkit #{0}] Left bulb's scale is {1}.", ModuleID, LogScale(bulb1Notes));
+        Debug.LogFormat("[The Cruel Modkit #{0}] Right bulb's scale is {1}.", ModuleID, LogScale(bulb2Notes));
+
+        for (int i = 0; i < 12; i++)
+        {
+            if (bulb1Actions[i] != 2)
+                bulb1Actions[i] = Random.Range(0, 2);
+            if (bulb2Actions[i] != 2)
+                bulb2Actions[i] = Random.Range(0, 2);
+        }
+        Debug.LogFormat("[The Cruel Modkit #{0}] Left bulb's key actions are {1}.", ModuleID, bulb1Actions.Select(x => new string[] { "Off", "On", "Toggle" }[x]).Join(", "));
+        Debug.LogFormat("[The Cruel Modkit #{0}] Right bulb's key actions are {1}.", ModuleID, bulb2Actions.Select(x => new string[] { "Off", "On", "Toggle" }[x]).Join(", "));
+    }
+    
+    public override void OnPianoPress(int Piano)
+    {
+        if (Module.IsAnimating())
+            return;
+        
+        Module.Audio.PlaySoundAtTransform(Module.PianoSounds[Piano + (Info.Piano * 12)].name, Module.transform);
+        Module.Piano[Piano].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+        
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.IsSolving())
+        {
+            if (!Module.CheckValidComponents())
+            {
+                Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The {1} key on the piano was pressed when the component selection was [{2}] instead of [{3}].", ModuleID, Info.PianoKeyNames[Piano], Module.GetOnComponents(), Module.GetTargetComponents());
+                Module.CauseStrike();
+                return;
+            }
+
+            Module.StartSolve();
+        }
+        if (Piano == lastPress & !BulbScrewedIn.Contains(false))
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! Pressed the {1} key twice in a row. Turning both bulbs off.", ModuleID, Info.PianoKeyNames[Piano]);
+            ChangeBulb(0, false);
+            ChangeBulb(1, false);
+            Module.CauseStrike();
+            return;
+        }
+
+        if (BulbScrewedIn[0] & BulbScrewedIn[1])
```

Comment:
```suggestion
        if (BulbScrewedIn[0] && BulbScrewedIn[1])
```
Use the lazy operator instead